source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -569,33 +596,38 @@ int StreamHandleWrapper::luaImportPublicKey(lua_State* state) {
   return 1;
 }
 
-FilterConfig::FilterConfig(const std::string& lua_code, ThreadLocal::SlotAllocator& tls,
-                           Upstream::ClusterManager& cluster_manager)
-    : cluster_manager_(cluster_manager), lua_state_(lua_code, tls) {
-  lua_state_.registerType<Filters::Common::Lua::BufferWrapper>();
-  lua_state_.registerType<Filters::Common::Lua::MetadataMapWrapper>();
-  lua_state_.registerType<Filters::Common::Lua::MetadataMapIterator>();
-  lua_state_.registerType<Filters::Common::Lua::ConnectionWrapper>();
-  lua_state_.registerType<Filters::Common::Lua::SslConnectionWrapper>();
-  lua_state_.registerType<HeaderMapWrapper>();
-  lua_state_.registerType<HeaderMapIterator>();
-  lua_state_.registerType<StreamInfoWrapper>();
-  lua_state_.registerType<DynamicMetadataMapWrapper>();
-  lua_state_.registerType<DynamicMetadataMapIterator>();
-  lua_state_.registerType<StreamHandleWrapper>();
-  lua_state_.registerType<PublicKeyWrapper>();
-
-  request_function_slot_ = lua_state_.registerGlobal("envoy_on_request");
-  if (lua_state_.getGlobalRef(request_function_slot_) == LUA_REFNIL) {
-    ENVOY_LOG(info, "envoy_on_request() function not found. Lua filter will not hook requests.");
+FilterConfig::FilterConfig(const envoy::extensions::filters::http::lua::v3::Lua& proto_config,
+                           ThreadLocal::SlotAllocator& tls,
+                           Upstream::ClusterManager& cluster_manager, Api::Api& api)
+    : cluster_manager_(cluster_manager) {
+  for (const auto& source_code : readSourceCodes(proto_config, api)) {
+    auto per_lua_code_setup_ptr = std::make_unique<PerLuaCodeSetup>(source_code.second, tls);
+    if (!per_lua_code_setup_ptr) {
+      continue;
+    }
+    per_lua_code_setups_map_[source_code.first] = std::move(per_lua_code_setup_ptr);
   }
+}
 
-  response_function_slot_ = lua_state_.registerGlobal("envoy_on_response");
-  if (lua_state_.getGlobalRef(response_function_slot_) == LUA_REFNIL) {
-    ENVOY_LOG(info, "envoy_on_response() function not found. Lua filter will not hook responses.");
+const absl::flat_hash_map<std::string, const std::string>
+FilterConfig::readSourceCodes(const envoy::extensions::filters::http::lua::v3::Lua& config,
+                              Api::Api& api) const {
+  absl::flat_hash_map<std::string, const std::string> source_codes;
+  if (!config.inline_code().empty()) {
```

Comment:
Optional. Not a blocker, but the config requires this, so we don't really need to check if this is empty.

Reply:
+1

Reply:
Also, this function is only used above, can you just inline it for clarity? This confused me on first read through. Seems like we can doaway with the intermediate map.